lib/script/taproot.js

```diff
@@ -0,0 +1,101 @@
+/*!
+* taproot.js - taproot interpreter for bcoin
+* https://github.com/bcoin-org/bcoin
+*/
+
+'use strict';
+
+const bio = require('bufio');
+const assert = require('assert');
+const { strcmp } = require('../utils/util');
+
+const {
+  TapLeafHash,
+  TapBranchHash,
+  TapTweakHash
+} = require('../utils/taggedhash');
+
+/**
+ * TapLeaf
+ */
+
+class TapLeaf {
+  /**
+   * Create a leaf in taproot tree.
+   * @constructor
+   * @param {Script} script
+   * @param {Number} leafVersion
+   * @param {String} name
+   */
+
+  constructor(script, leafVersion, name) {
+    assert(script);
+    assert(leafVersion);
+    this.script = script;
+    this.leafVersion = leafVersion;
+    this.name = name || '';
+  }
+
+  getLeafHash = () => {
+    const size = this.script.getVarSize();
+    const bw = bio.write(size + 1);
+    bw.writeU8(this.leafVersion);
+    bw.writeVarBytes(this.script.raw);
+    const data = bw.render();
+    return TapLeafHash.digest(data);
+  };
+}
+
+/**
+ * Create tree root from array of scripts
+ * @param {[TapLeaf]} scripts
+ * scripts: a list of items; each item is either:
+ *  - a TapLeaf
+ *  - another list of items (with the same structure)
+ * @returns {Buffer} tree root
+ */
+
+const taprootTreeHelper = (scripts) => {
+  if (scripts.length === 0) { // No script path
+    return null;
+  } else if (scripts.length === 1) { // Only one script, it's a leaf
```

Comment:
let's take out these comments, this one especially because it COULD be a leaf, but also could be an entire array...